From 662bc5b0d97576b712018596454664057d33ff69 Mon Sep 17 00:00:00 2001
From: Ronan Abhamon <ronan.abhamon@vates.fr>
Date: Mon, 14 Oct 2024 17:33:49 +0200
Subject: [PATCH] Fix override of FileSR.attach

The current attach method of FileSR doesn't correctly override the method of the SR class.
It actually adds a "bind" parameter, which is seen as an error by analyzers like mypy.

The "bind" parameter was added by this commit:
"CA-371791: Fix world readable permissions on EXTSR"

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>
---
 drivers/EXTSR.py     | 2 +-
 drivers/FileSR.py    | 5 ++++-
 tests/test_FileSR.py | 2 +-
 3 files changed, 6 insertions(+), 3 deletions(-)

Index: sm-3.2.12/drivers/EXTSR.py
===================================================================
--- sm-3.2.12.orig/drivers/EXTSR.py
+++ sm-3.2.12/drivers/EXTSR.py
@@ -123,7 +123,7 @@ class EXTSR(FileSR.FileSR):
                         'LVMMount',
                         opterr='FSCK failed on %s. Errno is %d' % (self.remotepath, inst.code))
 
-            super(EXTSR, self).attach(sr_uuid, bind=False)
+            self.attach_and_bind(sr_uuid, bind=False)
 
         self.attached = True
 
Index: sm-3.2.12/drivers/FileSR.py
===================================================================
--- sm-3.2.12.orig/drivers/FileSR.py
+++ sm-3.2.12/drivers/FileSR.py
@@ -156,7 +156,10 @@ class FileSR(SR.SR):
             raise xs_errors.XenError('FileSRDelete', \
                   opterr='error %d' % inst.code)
 
-    def attach(self, sr_uuid, bind=True):
+    def attach(self, sr_uuid):
+        self.attach_and_bind(sr_uuid)
+
+    def attach_and_bind(self, sr_uuid, bind=True):
         if not self._checkmount():
             try:
                 util.ioretry(lambda: util.makedirs(self.path, mode=0o700))
Index: sm-3.2.12/tests/test_FileSR.py
===================================================================
--- sm-3.2.12.orig/tests/test_FileSR.py
+++ sm-3.2.12/tests/test_FileSR.py
@@ -682,7 +682,7 @@ class TestFileSR(unittest.TestCase):
 
         sr.path = mount_dst
         sr.remotepath = mount_src
-        sr.attach(None, bind=False)
+        sr.attach_and_bind(None, bind=False)
 
         self.assertTrue(sr.attached)
 
